src/parser/inlines.rs

```diff
@@ -337,10 +343,11 @@ impl<'a, 'r, 'o, 'd, 'i, 'c, 'subj> Subject<'a, 'r, 'o, 'd, 'i, 'c, 'subj> {
                 let ix = match c.delim_char {
                     b'~' => 0,
                     b'^' => 1,
-                    b'"' => 2,
-                    b'\'' => 3,
-                    b'_' => 4,
-                    b'*' => 5 + (if c.can_open { 3 } else { 0 }) + (c.length % 3),
+                    b'=' => 2,
+                    b'"' => 3,
+                    b'\'' => 4,
+                    b'_' => 5,
+                    b'*' => 6 + (if c.can_open { 2 } else { 0 }) + (c.length % 3),
```

Comment:
I think we don't want to change this `3` to `2` here -- it causes an overlap in the `c.can_open` and `!c.can_open` cases. (Before, we had effectively `0+(c.length%3)` -- 0, 1, 2 -- and `3+(c.length%3)` -- 3, 4, 5. Now it looks like 0, 1, 2, and 2, 3, 4.)

Instead, we should keep this like this:

```suggestion
                    b'*' => 6 + (if c.can_open { 3 } else { 0 }) + (c.length % 3),
```

and add an extra element in the array here, by changing `11` to `12`:

https://github.com/kivikakk/comrak/blob/bb6ff755ba850e70b6477fc935a94f4da272dc57/src/parser/inlines.rs#L320